ozaria/site/common/ozariaUtils.js

```diff
@@ -30,18 +28,31 @@ export const findNextLevelsBySession = (sessions, levels, levelStatusMap) => {
     levelDataMap = levels
   }
   for (const [levelOriginal, level] of Object.entries(levelDataMap)) {
-    if (levelStatusMap[levelOriginal] === 'started') {
+    const levelStatus = levelStatusMap[levelOriginal]
+    const isLevelStarted = typeof levelStatus === 'string' && levelStatus === 'started'
+    const isLevelCompleted = (typeof levelStatus === 'string' && levelStatus === 'complete') || (typeof levelStatus === 'number' && levelStatus > 0)
+
+    if (isLevelStarted) {
       nextLevelOriginals.add(levelOriginal)
-    } else if (levelStatusMap[levelOriginal] === 'complete') {
-      const unlockedLevelOriginals = getNextLevelOriginalForLevel(level) || []
-      unlockedLevelOriginals
-        .filter((original) => levelStatusMap[original] !== 'complete')
-        .forEach((original) => nextLevelOriginals.add(original)) // incomplete unlocks
+    } else if (isLevelCompleted) {
+      let unlockedLevel = []
+      if (level.isPlayedInStages) {
+        const stageCompleted = levelStatusMap[levelOriginal]
+        unlockedLevel = getNextLevelForLevel(level, stageCompleted) || {}
+      } else {
+        unlockedLevel = getNextLevelForLevel(level) || {}
+      }
+      const unlockedLevelStatus = levelStatusMap[unlockedLevel.original]
```

Comment:
`unlockedLevel` could be an empty object literal `{}`. What does indexing `levelStatusMap[undefined]` return? Is this a possible situation?

Reply:
Yes that is possible if there is no next level for the given level. In that case, I believe `levelStatusMap[undefined]` would return undefined

Reply:
That appears correct. Thanks! =D